cockroachdb/goose/migration.go

```diff
@@ -0,0 +1,43 @@
+package utils
+
+import (
+	"context"
+	"database/sql"
+	"runtime"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
+	"github.com/pressly/goose/v3"
+)
+
+type PgxContextMigration func(context.Context, *pgx.Conn) error
+
+func AddPgxContextMigration(upMigration PgxContextMigration, downMigration PgxContextMigration) {
+	// runtime.Caller returns the filename of the migration file that calls this function
+	_, filename, _, ok := runtime.Caller(1)
+	if !ok {
+		panic("Could not get caller filename")
+	}
+	goose.AddNamedMigrationNoTxContext(
```

Comment:
question: why the `NoTx` version of the Goose functions and not `AddNamedMigrationContext`?

Reply:
using NoTX allows to have a DB instance instead of a transaction (sql.Tx).
It is essential to have an sql.DB to retrieve the pgx.Conn